### ISSUE TITLE
Only compact if on var

### ISSUE THREAD

**Issue by erikvader** [erikvader/MiniZinc-linter#31]
There is no point in rewriting `if`-expressions if it is `par` in rule #20.

**Comment by matsc-at-sics-se** [comment]
The bug is still there:

```
include "globals.mzn";

set of int: I = {2,3,5};

var 1..3: x;
var 1..3: y;
var 1..3: z;

constraint
  z = if card(I) > 1 then y else 0 endif;

constraint
  if card(I) > 1 then alldifferent([x,y,z]) else true endif;
```

```
$ lzn foo.mzn 
[...]
/home/matsc/foo.mzn:11.7-11.40: should be compacted [compacted-if(20)]
   |     z = if card(I) > 1 then y else 0 endif;
   |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
rewrite as: 
   |     (card(I) > 1)*y
```

**Comment by erikvader** [comment]
this particular case should now be fixed with 9fe16ea0110e0ab4077dee751f3c5dccdebe3ec5
